micrometer-core/src/main/java/io/micrometer/core/instrument/config/MeterFilter.java

```diff
@@ -52,7 +52,7 @@ static MeterFilter commonTags(Iterable<Tag> tags) {
         return new MeterFilter() {
             @Override
             public Meter.Id map(Meter.Id id) {
-                return id.withTags(Tags.concat(tags, id.getTagsAsIterable()));
+                return id.replaceTags(Tags.concat(tags, id.getTagsAsIterable()));
```

Comment:
If I'm understanding correctly, the motivation is to save one `Tags#concat()` call but get the same result. Is that correct? This is making me think we should have more tests and more clear JavaDocs about the behavior around common tags and maybe these withTags/replaceTags methods.

Reply:
@shakuzen Correct. I updated this to add some tests and update their Javadoc as suggested. Let me know what you think.

Reply:
For the record, I didn't update Javadoc for `withTags()` with its current behavior as I wasn't sure if it was intentional.

Reply:
We might also want to update Javadoc for sorting and deduplication logic in `Tags`.